Data/Csv/Encoding.hs

```diff
@@ -139,6 +140,13 @@ decodeByNameWith :: FromNamedRecord a
                  -> Either String (Header, Vector a)
 decodeByNameWith !opts = decodeWithP (csvWithHeader opts)
 
+-- | Should quoting be applied to text fields, and at which level
```

Comment:
Nit: remove "text". We don't look at what the field contains (it could be a number). This will (and should) apply to all fields.